src/rdkafka_sasl_oauthbearer_oidc.c

```diff
@@ -0,0 +1,397 @@
+        rd_http_error_t *herr;
+        char *jwt_token = NULL;
+        struct curl_slist *headers = NULL;
+        char *data_to_token;
+        size_t data_to_token_size;
+        const char *scope;
+        const char *token_url;
+        rd_bool_t build_headers_status;
+        char *error;
+
+        token_url = rk->rk_conf.sasl.oauthbearer.token_endpoint_url;
+        if (token_url == NULL) {
+                rd_kafka_log(rk, LOG_ERR, "OAUTH/OIDC",
+                             "token_url is a required field");
+                return;
+        }
+
+        build_headers_status = rd_false;
+        build_headers_status = build_headers(&rk->rk_conf, &headers, &error);
+        if (!build_headers_status) {
+                rd_kafka_log(rk, LOG_ERR, "OAUTH/OIDC",
+                             "Failed to build headers: %s",
+                             error);
+                rd_free(error);
+                return;
+        }
+
+        /* Build data field */
+        scope = rk->rk_conf.sasl.oauthbearer.scope;
+        if (scope == NULL) {
+                rd_kafka_log(rk, LOG_ERR, "OAUTH/OIDC",
+                             "scope is a required field");
+                return;
+        }
+        data_to_token_size = strlen(data) + strlen(scope) - 1;
+        data_to_token = rd_malloc(data_to_token_size);
+        rd_snprintf(data_to_token, data_to_token_size, data, scope);
+        herr = rd_http_post_json(token_url,
+                                 &headers,
+                                 data_to_token,
+                                 data_to_token_size,
+                                 timeout,
+                                 retry,
+                                 &json);
+        if (unlikely(herr != NULL)) {
+                rd_kafka_log(rk, LOG_ERR, "OAUTH/OIDC",
+                             "Failed to receive json result from HTTP(S), "
+                             "returned error code: %d, returned error string: "
+                             "%s", herr->code, herr->errstr);
+                rd_http_error_destroy(herr);
+                goto exit;
+        }
+        herr = rd_http_extract_jwt_from_json(&json, &jwt_token, token_key);
+        if (unlikely(herr != NULL)) {
+                rd_kafka_log(rk, LOG_ERR, "OAUTH/OIDC",
+                             "Failed to extract jwt from json, "
+                             "returned error code: %d, returned error string: "
+                             "%s", herr->code, herr->errstr);
+                rd_http_error_destroy(herr);
+                goto exit;
+        }
+
+        /* TODO: Forward token to broker. */
+
+exit:
+        RD_IF_FREE(data_to_token, rd_free);
+        cJSON_Delete(json);
+        RD_IF_FREE(headers, curl_slist_free_all);
+        RD_IF_FREE(jwt_token, rd_free);
+        return;
+}
+
+
+/**
+ * @brief make sure the jwt is able to be extracted from HTTP(S) requests.
+ */
+int unittest_sasl_oauthbearer_oidc_should_succeed (void) {
```

Comment:
use ut_ for the static sub-functions in the file itself. (also make this static)
use unittest_ for the single exported unittest function.